src/history/history.service.ts

```diff
@@ -206,7 +202,10 @@ export class HistoryService {
    */
   toHistoryEntryDto(entry: HistoryEntry): HistoryEntryDto {
     return {
-      identifier: entry.note.alias ? entry.note.alias : entry.note.id,
+      identifier:
+        !entry.note.aliases || entry.note.aliases.length === 0
+          ? entry.note.publicId
+          : entry.note.aliases.filter((alias) => alias.primary)[0].name,
```

Comment:
I wonder if we should extract this bit into a `getPrimaryAlias` method, to make clearer what happens.

Reply:
extracted.